src/main/java/com/coveo/feign/ReflectionErrorDecoder.java

```diff
@@ -135,7 +135,10 @@ private void initialize() {
       detailMessageField = Throwable.class.getDeclaredField("detailMessage");
       detailMessageField.setAccessible(true);
     } catch (Exception e) {
-      logger.debug("Unable to directly set the detailMessage, make sure exception do implement {}", baseExceptionClass.getName());
+      logger.debug(
+          "Unable to set the detailMessage via reflection, make sure the base exception do implement '{}'. Error message: '{}'.",
+          ExceptionMessageSetter.class.getName(),
```

Comment:
It's kind of weird that the class takes anything that extends exception (so is meant to be super flexible?). But will displays the `ExceptionMessageSetter` class name here. We could probably make it less flexible or respect the baseExceptionClass to avoid confusion.

As I think of it, it's a bit weird that `ExceptionMessageSetter` is kind of the implicit default ? Shouldn't the constructor use `ExceptionMessageSetter` as the base class if none is provided? (I think would also allow us to simplify some methods)

Reply:
The class cannot restrict the base exception as it parses exceptions from an interface `Class<?> apiClass` which cannot be typed and the exception declared on the methods are also out of the control of this class.

The constructor already takes a base exception that is typed (`S`).

`ExceptionMessageSetter` is just an interface and a way to get around the new limitations of jdk16+, it's not meant to be a base exception class.